src/server/routes/search.rs

```diff
@@ -151,51 +165,29 @@ async fn results(
             // default selected upstream search engines from the config file otherwise
             // parse the non-empty cookie and grab the user selected engines from the
             // UI and use that.
-            let mut results: SearchResults = match cookie_value {
-                Some(cookie_value) => {
-                    let engines: Vec<EngineHandler> = cookie_value
-                        .engines
-                        .iter()
-                        .filter_map(|name| EngineHandler::new(name).ok())
-                        .collect();
-
-                    match engines.is_empty() {
-                        false => {
-                            aggregate(
-                                query,
-                                page,
-                                config.aggregator.random_delay,
-                                config.debug,
-                                &engines,
-                                config.request_timeout,
-                                safe_search_level,
-                            )
-                            .await?
-                        }
-                        true => {
-                            let mut search_results = SearchResults::default();
-                            search_results.set_no_engines_selected();
-                            search_results
-                        }
-                    }
+            let mut results: SearchResults = match user_settings.engines.is_empty() {
+                false => {
```

Comment:
Using a match statement for a boolean. You probably want to use `if`

Reply:
> Using a match statement for a boolean. You probably want to use `if`

Thanks :heart: for taking the time to provide your code revew. We really appreciate it :+1: . :slightly_smiling_face: 

Well, we prefer having `match` over `if` because for code consistency. So we would suggest leaving it with the `match` approach. :slightly_smiling_face: .